AutoSteamApp/Process Memory/SteamworksData.cs

```diff
@@ -0,0 +1,235 @@
+        /// </summary>
+        ulong ButtonPressedCheckAddress;
+
+        /// <summary>
+        /// Address used to read what phase the steamworks is in. 
+        /// </summary>
+        ulong PhaseAddress;
+
+        /// <summary>
+        /// Address used to read what sub-phase the steamworks is in. When in the bonus or settled phase.
+        /// </summary>
+        ulong SecondPhaseAddress;
+
+        /// <summary>
+        /// Address used to read the rarity of the reward.
+        /// </summary>
+        ulong RarityAddress;
+
+        /// <summary>
+        /// Address used to read the current steam gauge progress.
+        /// </summary>
+        ulong SteamGaugeAddress;
+        
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// The value used to check whether the game has registered the input press.
+        /// </summary>
+        public byte InputPressStateCheck
+        {
+            get
+            {
+                return MemoryHelper.Read<byte>(MHWProcess, ButtonPressedCheckAddress);
+            }
+        }
+
+        /// <summary>
+        /// what phase the steamworks is in. 
+        /// </summary>
+        public byte PhaseValue
+        {
+            get
+            {
+                return MemoryHelper.Read<byte>(MHWProcess, PhaseAddress);
+            }
+        }
+
+        /// <summary>
+        /// what phase the steamworks is in when in the main menu of the steamworks.
+        /// </summary>
+        public byte SecondPhaseValue
+        {
+            get
+            {
+                return MemoryHelper.Read<byte>(MHWProcess, SecondPhaseAddress);
+            }
+        }
+
+        /// <summary>
+        /// what phase the steamworks is in when in the main menu of the steamworks.
+        /// </summary>
+        public RewardRarity RewardRarityValue
+        {
+            get
+            {
+                return (RewardRarity)MemoryHelper.Read<byte>(MHWProcess, RarityAddress);
+            }
+        }
+
+        /// <summary>
+        /// the current progress of the steam gauge.
+        /// </summary>
+        public short SteamGuageValue
+        {
+            get
+            {
+                return MemoryHelper.Read<short>(MHWProcess, SteamGaugeAddress);
+            }
+        }
+
+        #endregion
+
+        #region Constructor
+
+        public SteamworksData(Process mhwProcess)
+        {
+            Log.Debug("Loading Steamworks data values");
+            MHWProcess = mhwProcess;
+            LoadData();
+            Log.Debug("Steamworks data values loaded.");
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Loads/Reloads the Save Data of the current MHW Process
+        /// </summary>
+        public void LoadData()
+        {
+            // Load the address of the steamworks data
+            SteamworksAddress = MemoryHelper.Read<ulong>(MHWProcess, MHWMemoryValues.SteamworksDataPointer);
```

Comment:
It is a good practice to use empty lines to separate/differentiate statements with different purpose. This will improve readability and avoid the `wall of text, tl;dr` feeling